File: java/42586.java
import java.util.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        List<Integer> answer = new ArrayList<>();
        List<Integer> completeDays = new ArrayList<>(); // 완료 날짜만을 보관하는 리스트
        
        for (int i = 0; i < progresses.length; i++) {
            int left = 100 - progresses[i];
            int x = left / speeds[i];
            
            if (left % speeds[i] != 0) {
                x ++;
            }
            
            completeDays.add(x);
        }
        
        // System.out.println(completeDays);
        Queue<Integer> queue = new LinkedList<>();
        
        for (int i = 0; i < completeDays.size(); i++) {
            int tmp = completeDays.get(i);
            
            if (queue.isEmpty()) {
                queue.add(tmp);
                continue;
            }
            
            if (queue.peek() < tmp) {
                answer.add(queue.size());
                queue.clear();
            }
            
            queue.add(tmp);
        }
        
        
        answer.add(queue.size());
        return answer.stream().mapToInt(x->x).toArray();
    }
}

File: python/42586.py
def solution(progresses, speeds):
    answer = []
    # 1. 각 작업 당 배포까지 필요한 날을 구하기
    needDays = []
    
    for i, v in enumerate(progresses):
        # i == index, v == value
        tmp = v
        days = 0
        while tmp < 100:
            tmp = tmp + speeds[i]
            days = days + 1
        
        needDays.append(days)

    # 2. 일당 배포하는 기능 개수 구하기
    tmp = [] #list
    for i in needDays:
        if tmp and max(tmp) < i:
            answer.append(len(tmp))
            tmp.clear()
        
        tmp.append(i)
    
    answer.append(len(tmp))
    return answer